src/ci_workflow/ci_manifests.py

```diff
@@ -23,6 +26,19 @@ def __klass(filename: str) -> Union[Type[CiTestManifest], Type[CiInputManifest]]
         else:
             return CiInputManifest
 
+    @staticmethod
+    def __check_duplicate_component_names(file: TextIOWrapper) -> None:
+
+        yaml_dict = yaml.safe_load(file)
+        component_names = []
+        for component in yaml_dict['components']:
+            component_names.append(component['name'])
+
+        if set(Counter(component_names).values()) != set([1]):
+            raise ValueError(f"Duplicate components in manifest {file.name}")
```

Comment:
Is it possible to display which component is duplicated? Also can you add respective unit test cases for this method?

Reply:
I have added display of duplicate component names and unit tests, please have a look.